core/src/main/java/org/axonframework/deadline/quartz/DeadlineJob.java

```diff
@@ -122,8 +127,21 @@ private void executeScheduledDeadline(ScopeAwareProvider scopeAwareComponents,
      */
     public static class DeadlineJobDataBinder {
 
+        // Deprecated in favor of the separate DeadlineMessage keys | only maintained for backwards compatibility
+        @Deprecated
         private static final String SERIALIZED_DEADLINE_MESSAGE = "serializedDeadlineMessage";
+        // Deprecated in favor of the separate DeadlineMessage keys | only maintained for backwards compatibility
+        @Deprecated
         private static final String SERIALIZED_DEADLINE_MESSAGE_CLASS_NAME = "serializedDeadlineMessageClassName";
+        // Deadline Message JobDataMap keys
+        private static final String DEADLINE_NAME = "deadlineName";
+        private static final String DEADLINE_IDENTIFIER = "deadlineIdentifier";
+        private static final String DEADLINE_TIMESTAMP_EPOCH_MILLIS = "deadlineTimestamp";
+        private static final String SERIALIZED_DEADLINE_PAYLOAD = "serializedDeadlinePayload";
+        private static final String DEADLINE_PAYLOAD_CLASS_NAME = "serializedDeadlinePayloadClassName";
+        private static final String DEADLINE_PAYLOAD_REVISION = "serializedDeadlinePayloadRevision";
+        private static final String SERIALIZED_DEADLINE_METADATA = "serializedDeadlineMetadata";
+        // Deadline Scope JobDataMap keys
```

Comment:
This approach will fix the current issue. However, I've created a ticket #682 to address serialization/deserialization of messages in a more structural way.